numpy/core/tests/test_multiarray.py

```diff
@@ -5048,6 +5048,14 @@ def test_tofile_format(self):
             s = f.read()
         assert_equal(s, '1.51,2.00,3.51,4.00')
 
+    def test_tofile_cleanup(self):
+        x = np.zeros((10), dtype=object)
+        with open(self.filename, 'wb') as f:
+            assert_raises(IOError, lambda: x.tofile(f, sep=''))
+
+        # Dup-ed file handle should be closed or remove will fail.
+        os.remove(self.filename)
+
```

Comment:
Can you repeat this test with `x.tofile(self.filename)`, to check it closes both the python handle and the fd?

Reply:
This second check is now added.